internal/controllers/phase_ensure_operator_group.go

```diff
@@ -52,13 +54,14 @@ func (r *AddonReconciler) reconcileOperatorGroup(
 	currentOperatorGroup := &operatorsv1.OperatorGroup{}
 
 	err := r.Get(ctx, client.ObjectKeyFromObject(operatorGroup), currentOperatorGroup)
-	if errors.IsNotFound(err) {
+	if apierrors.IsNotFound(err) {
 		return r.Create(ctx, operatorGroup)
 	}
 	if err != nil {
 		return fmt.Errorf("getting OperatorGroup: %w", err)
 	}
 
+	// TODO can't we compare UID?
```

Comment:
Can you elaborate, please?
What would you compare the UID to and why?

Reply:
My question was can we avoid the deepEqual call? Maybe we should just do the `r.Update` call directly? I guess in this case it is not too bad because the OG spec is not terrible, but say we had a beefy spec like a CSV, this deepEqual becomes expensive.

Was thinking we could compare UID/revision, but realizing this is silly as only one object is "live" (again not sure what I was thinking there)

Reply:
I don't think there is any short cut making comparing easier.
You are right the check is not strictly necessary, the kube-apiserver is smart enough to do a no-op if nothing changes.
The reason for this client check is just to reduce load on the apiserver.

DeepEqual is not the most efficient, but speed is not a critical concern and we can do this easily while staying within our 100m core cpu limit :)